tensorflow/core/platform/cus.cc

```diff
@@ -28,32 +29,45 @@ float cus::castValueToF32(const uint32_t& u){
   return *(float*)&u;
 }
 
-tensorflow::cus CusAdd(tensorflow::cus a, tensorflow::cus b) {
+extern "C"{
+
+cus CusAdd(const cus a, const cus b) {
   return cus(static_cast<float>(a) + static_cast<float>(b));
 }
 
-tensorflow::cus CusSub(tensorflow::cus a, tensorflow::cus b) {
+cus CusSub(const cus a, const cus b) {
   return cus(static_cast<float>(a) - static_cast<float>(b));
 }
 
-tensorflow::cus CusMul(tensorflow::cus a, tensorflow::cus b) {
+cus CusMul(const cus a, const cus b) {
   return cus(static_cast<float>(a) * static_cast<float>(b));
 }
 
-tensorflow::cus CusDiv(tensorflow::cus a, tensorflow::cus b) {
+cus CusDiv(const cus a, const cus b) {
   return cus(static_cast<float>(a) / static_cast<float>(b));
 }
 
-tensorflow::cus CusNeg(tensorflow::cus a) {
+cus CusNeg(const cus a) {
   return cus(-static_cast<float>(a));
 }
 
-bool CusEq(tensorflow::cus a, tensorflow::cus b) { return static_cast<float>(a) == static_cast<float>(b); }
-bool CusNe(tensorflow::cus a, tensorflow::cus b) { return static_cast<float>(a) != static_cast<float>(b); }
-bool CusLt(tensorflow::cus a, tensorflow::cus b) { return static_cast<float>(a) < static_cast<float>(b);}
-bool CusLe(tensorflow::cus a, tensorflow::cus b) { return static_cast<float>(a) <= static_cast<float>(b);}
-bool CusGt(tensorflow::cus a, tensorflow::cus b) { return static_cast<float>(a) > static_cast<float>(b);}
-bool CusGe(tensorflow::cus a, tensorflow::cus b) { return static_cast<float>(a) >= static_cast<float>(b); }
+bool CusEq(const cus a, const cus b) { return static_cast<float>(a) == static_cast<float>(b); }
+bool CusNe(const cus a, const cus b) { return static_cast<float>(a) != static_cast<float>(b); }
+bool CusLt(const cus a, const cus b) { return static_cast<float>(a) < static_cast<float>(b);}
+bool CusLe(const cus a, const cus b) { return static_cast<float>(a) <= static_cast<float>(b);}
+bool CusGt(const cus a, const cus b) { return static_cast<float>(a) > static_cast<float>(b);}
+bool CusGe(const cus a, const cus b) { return static_cast<float>(a) >= static_cast<float>(b); }
+
+
+// if not force them, those two inline functions would not be compiled 
+cusToF32Type ForceCompileCastCusToF32(cus a) {
```

Comment:
why are you declaring CastCusToF32 as inline in the first place?

Reply:
because CastCusToF32 has the implementation in the header file. If it is not inlined, it will throw the multiple definition error

Reply:
why not move the definition to the .cc file?